### PR TITLE
superlu-dist: fix rocm variant for the master branch

### PR DESCRIPTION
**Pull request by liuyangzhuan**

### CODE REVIEW

File: var/spack/repos/builtin/packages/superlu-dist/package.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def append_from_variant(*args):
             if cuda_arch[0] != "none":
                 append_define("CMAKE_CUDA_ARCHITECTURES", cuda_arch[0])
 
-        if "+rocm" in spec and spec.satisfies("@amd"):
+        if "+rocm" in spec and (spec.satisfies("@amd") or spec.satisfies("@8.0.0:")):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/superlu-dist/package.py]: Curious.  Would "@8:" suffice?

**Comment by v-dobrev** [comment]
I tried this on a machine at LLNL similar to OLCF's Crusher and I get error that `mpi.h` cannot be found. Maybe I did not configure something right, or this may need update in SuperLU_DIST itself. Just wanted to let you know.

**Review by v-dobrev — DISMISSED** [review]

File: var/spack/repos/builtin/packages/superlu-dist/package.py (hunk 1 above)
Comment (liuyangzhuan) [reply var/spack/repos/builtin/packages/superlu-dist/package.py]: I've changed it to @8:

**Comment by v-dobrev** [comment]
I tested the last commit (0e35849) and it worked for me.

Note that the style check fails, so you need to fix that.

**Review by v-dobrev — DISMISSED** [review]

**Comment by xiaoyeli** [comment]
Thanks for reporting and fixing!

**Review by tldahlgren — DISMISSED** [review]

**Comment by tldahlgren** [comment]
@spackbot fix style

**Comment by eugeneswalker** [comment]
PIpelines are failing because of issues not related to the content of this PR. This is the issue of dependency binaries not being in the cache when they are expected to be there.
* [(specs) petsc/zw47q5w 3.18.1 oneapi@2022.1.0 linux-ubuntu20.04-x86_64 E4S OneAPI](https://gitlab.spack.io/spack/spack/-/jobs/4097757)
```
...
==> Error: No binary for python-3.8.13-vn4slfb74x2nj7a2wubesqvqbgwlcmcq found when cache-only specified
==> Error: Failed to install python due to SystemExit: 1
==> Error: python-3.8.13-vn4slfb74x2nj7a2wubesqvqbgwlcmcq: Package was not installed
==> Error: Installation request failed.  Refer to reported errors for failing package(s).
```

**Comment by eugeneswalker** [comment]
@spackbot run pipeline

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**